audtorch/datasets/speech_commands.py

```diff
@@ -0,0 +1,176 @@
+
+        files, targets = [], []
+        for speech_cmd in self.classes:
+            d = os.listdir(join(self.root, speech_cmd))
+            d = [join(speech_cmd, x) for x in d]
+
+            # Filter out test / train files using `testing_list.txt`
+            d_f = list(set(d) - set(test_files)) \
+                if train else list(set(d) & set(test_files))
+
+            files.extend([join(self.root, p) for p in d_f])
+            target = speech_cmd if speech_cmd in include else '_unknown_'
+            # speech commands is a classification dataset, so return logits
+            targets.extend([include.index(target) for _ in range(len(d_f))])
+
+        self.silence_label = len(include)
+
+        # Match occurrences of silence with `unknown`
+        # if silence:
+        #     n_samples = max(targets.count(len(include) - 1), 3000)
+        #     n_samples = int(n_samples * 0.9) \
+        #         if train else int(n_samples * 0.1)
+        #
+        #     sf = []
+        #     for file in os.listdir(join(self.root, '_background_noise_')):
+        #         if file.endswith('.wav'):
+        #             sf.append(join(self.root, '_background_noise_', file))
+        #
+        #     targets.extend([len(include) for _ in range(n_samples)])
+        #     files.extend(random.choices(sf, k=n_samples))
+
+        super().__init__(root, files, targets, sampling_rate,
+                         transform=transform,
+                         target_transform=target_transform)
+
+    def add_silence(self, n_samples=3000, same_length=True):
+        # https://github.com/audeering/audtorch/pull/49#discussion_r317489141
+        self.same_length = same_length
+        self.targets.extend([self.silence_label for _ in range(n_samples)])
+
+        bg_noises = []
+        for file in os.listdir(join(self.root, '_background_noise_')):
+            if file.endswith('.wav'):
+                bg_noises.append(join(self.root, '_background_noise_', file))
+
+        self.files.extend(random.choices(bg_noises, k=n_samples))
+
+    def __getitem__(self, index):
+        signal, signal_sampling_rate = load(self.files[index])
+        # Handle empty signals
+        if signal.shape[1] == 0:
+            warn('Returning previous file.', UserWarning)
+            return self.__getitem__(index - 1)
+        # Handle different sampling rate
+        if signal_sampling_rate != self.original_sampling_rate:
+            warn('Resample from {} to {}'
+                 .format(signal_sampling_rate, self.original_sampling_rate),
+                 UserWarning)
+            signal = resampy.resample(signal, signal_sampling_rate,
+                                      self.original_sampling_rate, axis=-1)
+
+        target = self.targets[index]
+
+        # https://github.com/audeering/audtorch/pull/49#discussion_r319044362
+        if target == self.silence_label and self.same_length:
+            trim = RandomCrop(self.sampling_rate)
```

Comment:
I guess it might be a little bit more efficient if you initialize the transform in the `init` part, even if they will be never used:
```
def __init__(self, root, ...):
    ...
    self.trim = RandomCrop(sampling_rate)

...

if target == self.silence_label and self.same_length:
    signal = self.trim(signal)
```

Reply:
Cool, good idea - implemented in https://github.com/audeering/audtorch/pull/49/commits/a36cab60e68bda64561c116dda5d71ef7609cf4c